src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs

```diff
@@ -843,6 +947,33 @@ private static ComWrappers.ComInterfaceEntry[] LookupVtableEntries(Type type)
                 source.AppendLine($$"""
                                                 return default;
                                             }
+                                    """);
+
+                if (hasRuntimeClasNameEntries)
+                {
+                    source.AppendLine($$"""
+                                private static string LookupRuntimeClassName(Type type)

```

Comment:
Nit: is the generated file with nullability annotations? If so, the return should be `string?`.

Reply:
Right now, it isn't.